ledger/participant-integration-api/src/main/scala/platform/apiserver/services/transaction/ApiTransactionService.scala

```diff
@@ -215,12 +203,24 @@ private[apiserver] final class ApiTransactionService private (
       .getTransactionById(transactionId, requestingParties)
       .map(getOrElseThrowNotFound)
 
-  @throws[StatusRuntimeException]
-  private def getOrElseThrowNotFound[A](a: Option[A]): A =
-    a.getOrElse(
-      throw Status.NOT_FOUND
-        .withDescription("Transaction not found, or not visible.")
-        .asRuntimeException()
-    )
+  private def transactionsLoggable(transactions: GetTransactionsResponse): String =
+    s"Responding with transactions: ${transactions.transactions.toList
+      .map(t => entityLoggable(t.commandId, t.transactionId, t.workflowId, t.offset))}"
+
+  private def transactionTreesLoggable(trees: GetTransactionTreesResponse): String =
+    s"Responding with transaction trees: ${trees.transactions.toList
+      .map(t => entityLoggable(t.commandId, t.transactionId, t.workflowId, t.offset))}"
 
+  private def entityLoggable(
+      commandId: String,
+      transactionId: String,
+      workflowId: String,
+      offset: String,
+  ): LoggingEntries =
+    LoggingEntries(
+      logging.commandId(commandId),
+      logging.transactionId(transactionId),
+      logging.workflowId(workflowId),
+      "offset" -> offset,
```

Comment:
How about adding a specialised function `String => LoggingEntry` in `logging` package for the offset?

Reply:
I think that's a good idea. I just moved this code but I'm happy to fix it up.